### PR TITLE
CMake: PNGwriter Version Support

### PR DESCRIPTION
**Pull request by ax3l**
Pull in third-party update: Adds version support to the PNGwriter's CMake module.

Follow-up to #1076, see https://github.com/ComputationalRadiationPhysics/cmake-modules/pull/11

--

Third party update for `FindmallocMC.cmake` in `thirdParty/cmake-modules` performed via

```bash
GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="picongpu@hzdr.de" \
    git subtree pull --prefix thirdParty/cmake-modules/ \
    git@github.com:ComputationalRadiationPhysics/cmake-modules.git dev --squash
```

### CODE REVIEW

**Comment by ax3l** [comment]
@Flamefire thanks for the reminder, this is the pull you are looking for.

File: thirdParty/cmake-modules/FindPNGwriter.cmake [hunk 1]
```diff
@@ -122,13 +122,49 @@ if(PNGwriter_FOUND)
...
+               PNGwriter_VERSION_DEFINE
+               REGEX "^[ \t]*#[ \t]*define[ \t]+PNGWRITER_VERSION[ \t]+")
+        if("${PNGwriter_VERSION_DEFINE}" MATCHES
+            "PNGWRITER_VERSION[ \t]+(0\\.[0-9])[0-9].*")
+            set(PNGwriter_VERSION "${CMAKE_MATCH_1}")
+        endif()
+        if("${PNGwriter_VERSION_DEFINE}" MATCHES
+            "PNGWRITER_VERSION[ \t]+0\\.[0-9]([0-9]).*")
+            set(PNGwriter_VERSION "${PNGwriter_VERSION}.${CMAKE_MATCH_1}")
+        endif()
```
Comment (Flamefire) [inline thirdParty/cmake-modules/FindPNGwriter.cmake]: Why checking 2 times for the same regex and not using 2 subgroups with 1 check?

Comment (psychocoderHPC) [inline thirdParty/cmake-modules/FindPNGwriter.cmake]: On the first call he greped the Minor version and in the second the patch level.

Comment (Flamefire) [inline thirdParty/cmake-modules/FindPNGwriter.cmake]: Yes, but why not grep both in one call?

Comment (ax3l) [inline thirdParty/cmake-modules/FindPNGwriter.cmake]: well, CMake style regex using multi-group did not work.

Comment (ax3l) [inline thirdParty/cmake-modules/FindPNGwriter.cmake]: as we just tested (thx @Flamefire !) it does indeed work. my clumsy fingers must have avoided a successful test last time. will update in the [according third party repo](https://github.com/ComputationalRadiationPhysics/cmake-modules) to a more slick version.

Comment (PrometheusPi) [inline thirdParty/cmake-modules/FindPNGwriter.cmake]: :+1:

**Pull request merged**